mlrun/utils/notifications/notification/mail.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import typing
+from email.message import EmailMessage
+
+import aiosmtplib
+
+import mlrun.common.schemas
+import mlrun.lists
+import mlrun.utils.helpers
+
+from .base import NotificationBase
+
+DEFAULT_SMTP_PORT = 587
+
+
+class MailNotification(NotificationBase):
+    """
+    API/Client notification for sending run statuses as a mail message
+    """
+
+    required_params = [
+        "server_host",
+        "server_port",
+        "sender_address",
+        "username",
+        "password",
+        "email_addresses",
+    ]
+
+    @classmethod
+    def validate_params(cls, params):
+        for required_param in cls.required_params:
+            if required_param not in params:
+                raise ValueError(
+                    f"Parameter '{required_param}' is required for MailNotification"
+                )
+
+        if type(params["email_addresses"]) not in [str, list]:
+            raise ValueError(
+                "Parameter 'email_addresses' must be a string or a list of strings"
+            )
```

Comment:
By this code below it seems that `email_addresses` is always a string.

```
      if type(email_addresses) is list:
            email_addresses = ",".join(email_addresses)
        params["email_addresses"] = email_addresses
        ```

Reply:
The `validate_params` function is called before the `enrich_default_params`.
So first we want to validate that the `email_addresses` is either string or list, and ,later we want to enrich it to string.